modules/apps/27-interchain-accounts/types/account.go

```diff
@@ -23,7 +23,7 @@ func GenerateAddress(moduleAccAddr sdk.AccAddress, portID string) sdk.AccAddress
 
 // ParseAddressFromVersion trims the interchainaccounts version prefix and returns the associated account address
 func ParseAddressFromVersion(version string) string {
-	return strings.TrimPrefix(version, fmt.Sprint(VersionPrefix, Delimiter))
+	return strings.Split(version, Delimiter)[1]
```

Comment:
The version string validation in `OnChanOpenTry` is such that this should never encounter an index out of bounds but I'm thinking it's probably worth adding a defensive check here regardless